batteries/argparse.luau

```diff
@@ -0,0 +1,137 @@
+local argparse = {}
+argparse.__index = argparse
```

Comment:
Check out `time.luau` in `std` for how to do the typing for this file generally, I've already done it with `duration` there. Should be something like:

```
type ModuleData = {
    -- whatever fields we're calling the data for the core type here
}

type ModuleInterface = typeof(argparse)
export type Module = setmetatable<ModuleData, ModuleInterface>
```